xaitk_saliency/utils/tools/gen_detector_sal.py

```diff
@@ -0,0 +1,137 @@
+import numpy as np
+from typing import Optional, Collection, Iterable, Tuple, Dict, Hashable
+
+from smqtk_detection.interfaces.detect_image_objects import DetectImageObjects
+from smqtk_detection.utils.bbox import AxisAlignedBoundingBox
+
+from xaitk_saliency.interfaces.gen_detector_prop_sal import GenerateDetectorProposalSaliency
+from xaitk_saliency.interfaces.perturb_image import PerturbImage
+from xaitk_saliency.utils.masking import occlude_image_batch
+from xaitk_saliency.utils.detection import format_detection
+
+
+def gen_detector_sal(
+    ref_imgs: Collection[np.ndarray],
+    detector: DetectImageObjects,
+    img_perturber: PerturbImage,
+    sal_generator: GenerateDetectorProposalSaliency,
+    verbose: Optional[bool] = False,
+) -> Collection[np.ndarray]:
+    """
+    Generate saliency maps for detections in each of the given images.
+    Detections generation, image perturbation, and saliency map creation is
+    performed by the implemenations of ``DetectImageObjects``,
+    ``PerturbImage``, and ``GenerateDetectorProposalSaliency`` that are passed,
+    respectively.
+
+    The resulting saliency maps matrix for each image has a shape of
+    [nDets, imgHeight, imgWidth].
+
+    :param ref_imgs: List of images to generate saliency maps for.
+    :param detector: Implementation of ``DetectImageObjects`` to generate
+        detections with.
+    :param img_perturber: Implementation of ``PerturbImage`` to use.
+    :param sal_generator: Implementation of
+        ``GenerateDetectorProposalSaliency`` to use.
+    :param verbose: Option to print out progress statements. Default is false.
+
+    :returns: List of saliency map matrices, each with shape
+        [nDets, imgHeight, imgWidth] respective to their corresponding image.
+    """
+
+    def dets_to_formatted_mat(
+        dets: Iterable[Tuple[AxisAlignedBoundingBox, Dict[Hashable, float]]],
+        class_keys: list
+    ) -> np.ndarray:
+        """
+        This helper function takes detections as returned by an implementation
+        of DetectImageObjects and returns a matrix formatted for use with the
+        DRISE algorithm, as returned by the format_detection() function.
+        """
+
+        bbox_mat = np.asarray([
+            [
+                bbox.min_vertex[0],
+                bbox.min_vertex[1],
+                bbox.max_vertex[0],
+                bbox.max_vertex[1]
+            ] for bbox in [det[0] for det in dets]
+        ])
+
+        if bbox_mat.shape[0] == 0:
+            return np.asarray([])
+
+        class_dicts = [det[1] for det in dets]
+
+        class_mat = np.asarray([
+            [class_dict[key] for key in class_keys]
+            for class_dict in class_dicts
+        ])
+
+        return format_detection(bbox_mat, class_mat)
+
+    if verbose:
```

Comment:
We can reduce the quantity if `if verbose` lines here by assigning to a lambda a wrapper around the logging function, or a function that drops the input message on the floor:
```
if verbose:
    _log = LOG.info
else:
    _log = lambda _: None
```

We should also be using logging here instead of prints, e.g. create on the module level:
```
import logging
LOG = logging.getLogger(__name__)
```